### PR TITLE
Introduce `validators` field in kustomization file

### PR DESCRIPTION
**Pull request by Shell32-Natsu**
close https://github.com/kubernetes-sigs/kustomize/issues/2604

### CODE REVIEW

File: api/internal/plugins/execplugin/execplugin.go [hunk 1]
```diff
@@ -140,6 +140,11 @@ func (p *ExecPlugin) Transform(rm resmap.ResMap) error {
 	return p.updateResMapValues(output, rm)
 }
 
+func (p *ExecPlugin) Validate(rm resmap.ResMap) error {
+	// Validate works exactly same with Transformer
```
Comment (monopole) [inline api/internal/plugins/execplugin/execplugin.go:144]: I'd rather defer introducing a new interface for now.

Once we make it, it's forever, and we're  not sure this  is a good idea.

In particular, the validator interface is identical to the  existing transformer interface - which suggests something is amiss. I suspect we'll move to three new interfaces more closely aligned with Config Functions, i.e. something like

```
 Generate(config) (ResourceList, error)
 Validate(config, ResourceList) error
 Transform(config, ResourceList) (ResourceList, error)
```
For now, the issue is only about adding a new field to run transformers -
specifically existing transformers - in a different context, as final step where 
they can make only limited changes.
Let's see if that gets used before formalizing a new interface.

File: api/internal/plugins/loader/loader.go [hunk 2]
```diff
@@ -87,6 +87,32 @@ func (l *Loader) LoadTransformer(
...
+	if err != nil {
+		return nil, err
+	}
+	t, ok := c.(resmap.Validator)
+	if !ok {
+		return nil, fmt.Errorf("plugin %s not a validator", res.OrgId())
+	}
+	return t, nil
+}
+
```
Comment (monopole) [inline api/internal/plugins/loader/loader.go:115]: So these would go away with no new interface.
Just maintain some var `validators []resmap.Transformer`

File: api/internal/target/kusttarget.go [hunk 3]
```diff
@@ -278,6 +282,54 @@ func (kt *KustTarget) configureExternalTransformers() ([]resmap.Transformer, err
...
+		delete(labels, validatedByLabelName)
+		if len(labels) == 0 {
+			r.SetLabels(nil)
+		} else {
+			r.SetLabels(labels)
+		}
+	}
+}
+
+func (kt *KustTarget) configureExternalValidators() ([]resmap.Validator, error) {
```
Comment (monopole) [inline api/internal/target/kusttarget.go:324]: suggest modifying `configureExternalTransformers` to accept a field name, and re-use it to load validators and transformers.

File: api/internal/target/kusttarget.go [hunk 4]
```diff
@@ -278,6 +282,54 @@ func (kt *KustTarget) configureExternalTransformers() ([]resmap.Transformer, err
...
+		kt.removeValidatedByLabel(new)
+		if err = orignal.ErrorIfNotEqualSets(new); err != nil {
+			return fmt.Errorf("validator shouldn't modify the resource map: %v", err)
+		}
+	}
+	return nil
+}
+
+func (kt *KustTarget) removeValidatedByLabel(rm resmap.ResMap) {
+	var validatedByLabelName string = "validated-by"
```
Comment (monopole) [inline api/internal/target/kusttarget.go:307]: declare in api/konfig/general.go like `managed-by`

**Comment by Shell32-Natsu** [comment]
@monopole Updated

**Pull request merged**